crates/router/src/connector/utils.rs

```diff
@@ -145,7 +151,9 @@ where
                 let captured_amount =
                     types::Capturable::get_captured_amount(&self.request, payment_data);
                 let total_capturable_amount = payment_data.payment_attempt.get_total_amount();
-                if Some(total_capturable_amount) == captured_amount {
+                if total_capturable_amount
```

Comment:
This condition will fail is total_capturable_amount is 0 and captured_amount is None.

Reply:
if captured amount it will default to 0 right, so the comparison will be between 

let's say total capturable amount is 100

`MinorUnit(100) == MinorUnit(0)` which will move it to else condition